framework/bin/gen_tests.pl

```diff
@@ -0,0 +1,284 @@
+use Utils;
+use Project;
+use Log;
+
+#
+# Process arguments and issue usage message if necessary.
+#
+my %cmd_opts;
+getopts('g:p:v:o:n:b:t:D', \%cmd_opts) or pod2usage(1);
+my $TOOL = $cmd_opts{g};
+# Print all supported generators, regardless of the other arguments, if -g help
+# is set
+defined $TOOL and $TOOL =~ /^help$/ and is_generator_valid($TOOL) || exit 1;
+
+pod2usage(1) unless defined $cmd_opts{g} and
+                    defined $cmd_opts{p} and
+                    defined $cmd_opts{v} and
+                    defined $cmd_opts{n} and
+                    defined $cmd_opts{b} and
+                    defined $cmd_opts{o};
+
+# Check whether the requested test generator is valid
+is_generator_valid($TOOL) || exit 1;
+
+my $PID = $cmd_opts{p};
+# Instantiate project
+my $project = Project::create_project($PID);
+
+my $VID = $cmd_opts{v};
+# Verify that the provided version id is valid
+my $BID = Utils::check_vid($VID)->{bid};
+$project->contains_version_id($VID) or die "Version id ($VID) does not exist in project: $PID";
+
+my $TID = $cmd_opts{n};
+$TID =~ /^\d+$/ or die "Wrong test_id format (\\d+): $TID!";
+my $TIME = $cmd_opts{b};
+$TIME =~ /^\d+$/ or die "Wrong budget format (\\d+): $TIME!";
+my $OUT_DIR = $cmd_opts{o};
+
+# Enable debugging if flag is set
+$DEBUG = 1 if defined $cmd_opts{D};
+
+# List of loaded classes
+my $LOADED_CLASSES = "$SCRIPT_DIR/projects/$PID/loaded_classes/$BID.src";
+
+# List of modified classes
+my $MOD_CLASSES = "$SCRIPT_DIR/projects/$PID/modified_classes/$BID.src";
+
+# Temporary directory for project checkout
+my $TMP_DIR = Utils::get_tmp_dir($cmd_opts{t});
+system("mkdir -p $TMP_DIR");
+
+# Set working directory
+$project->{prog_root} = $TMP_DIR;
+
+=pod
+
+=head2 Logging
+
+By default, the script logs all errors and warnings to F<gen_tests.log> in
+the temporary project root.
+Upon success, the log of this script is appended to:
+F<out_dir/logs/C<project_id>.C<version_id>.log>.
+
+=cut
+# Log file in output directory
+my $LOG_DIR = "$OUT_DIR/logs";
+my $LOG_FILE = "$LOG_DIR/$PID.$VID.log";
+system("mkdir -p $LOG_DIR");
+
+# Checkout and compile project
+$project->checkout_vid($VID) or die "Cannot checkout!";
+$project->compile() or die "Cannot compile!";
+
+# Open temporary log file
+my $LOG = Log::create_log("$TMP_DIR/$PID.$VID.$TID.log");
+$LOG->log_time("Start test generation");
+$LOG->log_msg("Parameters:");
+$LOG->log_msg(" -g $TOOL");
+$LOG->log_msg(" -p $PID");
+$LOG->log_msg(" -v $VID");
+$LOG->log_msg(" -n $TID");
+$LOG->log_msg(" -b $TIME");
+
+my $cp_file = "$TMP_DIR/project.cp";                               
+$project->_ant_call("export.cp.compile", "-Dfile.export=$cp_file")
+        or die "Cannot determine project classpath";
+
+# Export all environment variables that are expected by the wrapper script of
+# the test generator.
+$ENV{D4J_HOME}                = "$BASE_DIR";
+$ENV{D4J_FILE_ALL_CLASSES}    = "$LOADED_CLASSES";
+$ENV{D4J_FILE_TARGET_CLASSES} = "$MOD_CLASSES";
```

Comment:
You might want to rename this variable to `D4J_FILE_MOD_CLASSES` or `D4J_FILE_MODIFIED_CLASSES`. I find the name `_TARGET_` in this context quite confusing.